### PR TITLE
Api prototype

### PR DESCRIPTION
**Pull request by RolandDoll**
copied into a fresh branch off main as requested.

### CODE REVIEW

**Review by lawhorkl — CHANGES_REQUESTED** [review]
My only review so far is this:

How do you plan to scale the application in terms of folder structure? In real-world APIs your API program could have thousands of different endpoints/files so where do all those definitions go? Inside of index?

I would like to see a structured project with all TS files grouped together like
`routes/`, `services/`, etc.

**Comment by RolandDoll** [comment]
Endpoints have gone into their own file. One generic function handles all routes

File: index.ts [hunk 1]
```diff
@@ -2,35 +2,29 @@
...
 };
 
-process.on('unhandledRejection', (err) => {
-
+process.on('unhandled rejection', (err) =>
+{
     console.log(err);
     process.exit(1);
-});
+})
```
Comment (lawhorkl) [inline index.ts]: changing `unhandledRejection` probably makes this code block not work

File: src/endpoints/todoEndpoints.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import {Tasks} from "../models/ArrayModel"
+export default [
+    
+    // Get all
+    {
+        method: 'GET',
+        path: '/tasks',
+        handler: (
+            request: any, 
+            h: any) =>
```
Comment (lawhorkl) [inline src/endpoints/todoEndpoints.ts]: Part of the advantage of using typescript is being able to strongly type variables. Hapi has decent typescript support you need to do some research regarding using Hapi TypeScript types

**Review by lawhorkl — CHANGES_REQUESTED** [review]

**Comment by RolandDoll** [comment]
"any" typing has been replaced with specific typing.

**Review by lawhorkl — APPROVED** [review]

**Pull request merged**